src/main/java/com/amazon/opendistroforelasticsearch/jobscheduler/scheduler/JobScheduler.java

```diff
@@ -145,6 +146,19 @@ boolean reschedule(ScheduledJobParameter jobParameter, JobSchedulingInfo jobInfo
             return true;
         }
         Duration duration = Duration.between(this.clock.instant(), nextExecutionTime);
+
+        Instant secondExecutionTimeFromNow = jobParameter.getSchedule().getNextExecutionTime(nextExecutionTime);
+        if(secondExecutionTimeFromNow != null) {
+            Duration interval = Duration.between(nextExecutionTime, secondExecutionTimeFromNow);
+            if (interval.toMillis() > 0) {
+                double jitter = jobParameter.getJitter() == null ? 0d : jobParameter.getJitter();
+                jitter = jitter > 1 ? 1 : jitter;
+                jitter = jitter < 0 ? 0 : jitter;
```

Comment:
I suppose `jitter` is the percentage here, then it should definitely less than 1 (100%), otherwise, it may cause skipped execution.  maybe we want to define the max to be something like (95%)

Reply:
Good point, will add dynamic setting for jitter upper limit.